src/ui/public/vislib/lib/axis_labels.js

```diff
@@ -5,37 +5,17 @@ import ErrorHandlerProvider from 'ui/vislib/lib/_error_handler';
 export default function AxisLabelsFactory(Private) {
 
   const ErrorHandler = Private(ErrorHandlerProvider);
-  const defaults = {
-    show: true,
-    rotate: 0,
-    rotateAnchor: 'center',
-    filter: false,
-    color: '#ddd',
-    font: '"Open Sans", "Lato", "Helvetica Neue", Helvetica, Arial, sans-serif', // TODO
-    fontSize: '8pt',
-    truncate: 100
-  };
-
-  /**
-   * Appends axis title(s) to the visualization
-   *
-   * @class AxisLabels
-   * @constructor
-   * @param el {HTMLElement} DOM element
-   * @param xTitle {String} X-axis title
-   * @param yTitle {String} Y-axis title
-   */
 
   class AxisLabels extends ErrorHandler {
-    constructor(axis, attr) {
+    constructor(config, scale) {
       super();
-      _.extend(this, defaults, attr);
-      this.axis = axis;
+      this.config = config;
+      this.axisScale = scale;
 
       // horizontal axis with ordinal scale should have labels rotated (so we can fit more)
-      if (this.axis.isHorizontal() && this.axis.scale.isOrdinal()) {
-        this.filter = attr && attr.filter ? attr.filter : false;
-        this.rotate = attr && attr.rotate ? attr.rotate : 70;
+      if (this.config.isHorizontal() && this.config.isOrdinal()) {
+        this.filter = config.labels && config.labels.filter ? config.labels.filter : false;
```

Comment:
these could be updated to use the `config.get()` function with the default argument:

```js
this.filter = config.get('labels.filter', false)
```